### ISSUE TITLE
[Bug] Support list-based boolean masks for `TensorDict`

### ISSUE THREAD

**Issue by nicolas-dufour** [facebookresearch/rl#258]
The TensorDict doesn't support masking with lists whereas torch.Tensor works

Exemple:

```
td = TensorDict({"a": torch.randn(3,4,2), "b": torch.randn(3,4)}, [3, 4])

mask = [True, False, True]

td[mask]
```
raises `StopIteration` whereas `torch.randn(3,4)[mask]` works fine

**Comment by vmoens** [comment]
Thanks for reporting this @nicolas-dufour. I guess we could transform all index indices in tensors:
```python
    def __getitem__(self, index):
        if isinstance(index, list):
            index = torch.tensor(index, device=self.device)
```
and same for all `__getitem__` and `__setitem__` in `tensordict.py`.

That is also true if one element of a tuple is a list:
```python
x = torch.randn(2, 15)
x[[True, False], 2]
```
here only the list should be mapped to a tensor, hence we need something like (but cleaner):
```python
    def __getitem__(self, index):
        if isinstance(index, tuple) and any(isinstance(sub_index, list) for sub_index in index):
            index = tuple(torch.tensor(sub_index, device=self.device) if isinstance(sub_index, list) else sub_index for sub_index in index)
```


We should also update the tests to cover these use cases.
